conjur/user/user_logic.py

```diff
@@ -34,48 +34,50 @@ def rotate_api_key(self, resource_to_update):
         1. The user can either rotate their own API key or
         2. Rotate another users API key
         """
-        loaded_credentials = self.extract_credentials_from_credential_store()
-        user_from_credential_store = loaded_credentials['login_id']
-        if resource_to_update == user_from_credential_store:
-            raise InvalidOperationException("To rotate the API key of the currently logged-in user "\
-                                   "use this command without any flags or options")
-        if resource_to_update is not None:
-            user_from_credential_store = resource_to_update
-            logging.debug(f"Rotating API key for '{user_from_credential_store}'")
-
-            new_api_key = self.rotate_other_api_key(user_from_credential_store)
-            logging.debug(f"Successfully rotated API key for '{user_from_credential_store}'")
-        else:
-            # if the user does not provide a user to rotate their API key,
-            # their own API key will be rotated
+        logged_in_credentials = self.extract_credentials_from_credential_store()
+        logged_in_username = logged_in_credentials['login_id']
+
+        # if the user provides their own user id and they are logged in,
+        # then for the sake of a successful outcome, we will ignore the input.
+        # Otherwise, the Conjur server will fail the request.
+        # if the user does not provide a user to rotate their API key,
+        # their own API key will be rotated.
```

Comment:
This comment reminds me of something @jonahx said to me once:
> I propose we call this a "Pinocchio" comment.
Just like Pinocchio was a puppet who wanted to be real boy, sometimes comments want to be a real variable name.

In your case, this comment want to be a real method...
Extract the inner code of this `if` to a method with proper name, and most of the comment will be redundant.

I think that the `else` deserve a method too 😉